src/sgame/sg_bot.cpp

```diff
@@ -393,6 +393,18 @@ void G_BotThink( gentity_t *self )
 	//reset command buffer
 	usercmdClearButtons( botCmdBuffer->buttons );
 
+	if ( BG_Class( self->client->ps.stats[STAT_CLASS] )->abilities & SCA_WALLCLIMBER )
+	{
+		//wall walkers really want to wall walk when possible. Naive, ugly, but if it could work... worth it.
```

Comment:
> wall walkers really want to wall walk when possible

Far from true. In my experience having wallwalk on when you are trying to move around on the ground interferes with locomotion a lot. Like when you just want to jump you will stick to things when landing. Or you start to go up walls when trying to walk near them.

Reply:
The point is, this code is for bots, they won't loose orientation as humans when the world rotate like crazy. Unlike human, bots require navmesh to be generated to wallwalk, or to have manual connections added between navmeshes.
And in combat, they simply ignore navmesh stuff and thus would not be able to use the wallwalking ability.

The way I did it (which is, as my comment says, naive and ugly) allows AI to wallwalk in combat, even if this will be rare for the reasons you mentioned: it will only apply when bots pounce or jump but miss their target and end up on a wall.
I have not watched lot of games since I implemented this, and so I only witnessed this happen once.

Reply:
Sounds like more testing is needed.